be-typescript-express quickstarter first build after provisioning fails

**Describe the bug**
be-typescript-express component first build after provisioning fails at `build` stage

**To Reproduce**
Steps to reproduce the behavior:
1. In ODS 2.x/master, go to provisioning app and provision be-typescript-express component
2. Provision works
3. First job after repo/component is created fails

**Expected behavior**
 The component should be provisioned without issues

**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack: master, 2.x (might be any version)

**Log Output (ensure to remove any confidential information like tokens, project names, etc.**
```bash
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] sh
+ npm install
npm WARN be-ts-express@0.0.0 No repository field.

added 396 packages from 600 contributors in 4.687s
[Pipeline] sh
+ npm run build

> be-ts-express@0.0.0 build /tmp/workspace/newproject-cd/newproject-cd-be-ts-express-master
> npm run clean && npm run lint && echo Using TypeScript && tsc --version && tsc --pretty


> be-ts-express@0.0.0 clean /tmp/workspace/newproject-cd/newproject-cd-be-ts-express-master
> rimraf lib


> be-ts-express@0.0.0 lint /tmp/workspace/newproject-cd/newproject-cd-be-ts-express-master
> tslint --force --format verbose "src/**/*.ts"

Using TypeScript
Version 2.9.2
node_modules/@types/connect/index.d.ts:21:42 - error TS2689: Cannot extend an interface 'http.IncomingMessage'. Did you mean 'implements'?

21     export class IncomingMessage extends http.IncomingMessage {
                                            ~~~~~~~~~~~~~~~~~~~~


npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! be-ts-express@0.0.0 build: `npm run clean && npm run lint && echo Using TypeScript && tsc --version && tsc --pretty`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the be-ts-express@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/jenkins/.npm/_logs/2020-01-22T09_34_35_910Z-debug.log
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // wrap
[Pipeline] echo
***** Finished ODS Pipeline for  be-ts-express (with error) *****
[Pipeline] echo
Setting BitBucket build status to FAILED ...
[Pipeline] withCredentials
Masking only exact matches of $USERPASS
[Pipeline] {
[Pipeline] sh
+ curl --fail --silent --user '****' --request POST --header 'Content-Type: application/json' --data '{"state":"FAILED","key":"newproject-cd/gernd-cd-be-ts-express-master-1-45e083ee","name":"newproject-cd/newproject-cd-be-ts-express-master-1-45e083ee","url":"https://jenkins-newproject-cd.22ad.bi-x.openshiftapps.com/job/newproject-cd/job/newproject-cd-be-ts-express-master/1/"}' https://bitbucket.bix-digital.com/rest/build-status/1.0/commits/xxxxx
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] emailextrecipients
[Pipeline] emailext
An attempt to send an e-mail to empty list of recipients, ignored.
[Pipeline] }
[Pipeline] // node
[Pipeline] }
[Pipeline] // podTemplate
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE
```

--- a/be-typescript-express/Jenkinsfile
+++ b/be-typescript-express/Jenkinsfile
@@ -9,8 +9,8 @@ def group = env.GROUP_ID
 
 def credentialsId = "${projectId}-cd-cd-user-with-password"
 def quickstarterBasepath = "be-typescript-express"
-def yoVersion = "3.0.0"
-def generatorVersion = "0.5.1"
+def yoVersion = "3.1.1"
+def generatorVersion = "4.2.2"
 def dockerRegistry
 def gitHost
 // vars from jenkins master

--- a/be-typescript-express/Jenkinsfile.template
+++ b/be-typescript-express/Jenkinsfile.template
@@ -49,8 +49,8 @@ def stageBuild(def context) {
 
 def stageUnitTest(def context) {
   stage('Unit Test') {
-    sh "npm test -- --progress false"
-    sh "npm run coverage"
+    sh "npm test"
+    sh "npm run coverage -- --forceExit"
     if (false) {
       junit 'artifacts/xunit.xml'
       step([

--- a/be-typescript-express/fix/routes/weather.ts
+++ b/be-typescript-express/fix/routes/weather.ts
@@ -1,0 +1,13 @@
+import {Request, Response, Router} from "express";
+const weatherRouter: Router = Router();
+
+weatherRouter.post("/", (req, res) => {
+    // handle request here
+    res.sendStatus(200);
+});
+weatherRouter.get("/", (req, res) => {
+    // handle request here
+    res.send("You are my sunshine!");
+});
+
+export default weatherRouter;

--- a/docs/antora.yml
+++ b/docs/antora.yml
@@ -1,6 +1,6 @@
 name: ods-quickstarters
 title: ODS Quickstarters
-version: latest
+version: 2.0
 nav: 
 - modules/ROOT/nav.adoc
 

